pkg/cmd/roachtest/test_runner.go

```diff
@@ -744,13 +744,15 @@ func (r *testRunner) runTest(
 		}
 	}
 
-	// Detect replica divergence (i.e. ranges in which replicas have arrived
-	// at the same log position with different states).
-	c.FailOnReplicaDivergence(ctx, t)
-	// Detect dead nodes in an inner defer. Note that this will call
-	// t.printfAndFail() when appropriate, which will cause the code below to
-	// enter the t.Failed() branch.
-	c.FailOnDeadNodes(ctx, t)
+	if !t.Failed() {
+		// Detect replica divergence (i.e. ranges in which replicas have arrived
+		// at the same log position with different states).
+		c.FailOnReplicaDivergence(ctx, t)
```

Comment:
Can this one always run? Tests that fail are potentially more interesting targets to check for divergence. Note that this method was written to tolerate unhealthy clusters; it won't cause additional failures (or shouldn't, anyway).

Reply:
:+1: